src/components/password/password.spec.tsx

```diff
@@ -0,0 +1,201 @@
+import React from "react";
+import { mount, ReactWrapper } from "enzyme";
+import { act } from "react-dom/test-utils";
+import Password, { PasswordProps } from "./password.component";
+import { assertStyleMatch } from "../../__spec_helper__/test-utils";
+import { HiddenAriaLive } from "./password.style";
+import StyledButtonMinor from "../button-minor/button-minor.style";
+import Label from "../../__internal__/label";
+import Textbox from "../textbox";
+import guid from "../../__internal__/utils/helpers/guid";
+
+const mockedGuid = "guid-12345";
+jest.mock("../../__internal__/utils/helpers/guid");
+
+(guid as jest.MockedFunction<typeof guid>).mockImplementation(() => mockedGuid);
+
+function renderPasswordInput(props: PasswordProps) {
+  return mount(<Password {...props} />);
+}
+
+describe("Password Input", () => {
+  let wrapper: ReactWrapper;
+  let input: ReactWrapper;
+
+  const mockEvent = {
+    nativeEvent: {
+      stopImmediatePropagation: () => {},
+    },
+  };
+
+  describe("when rendered", () => {
+    it("should have the Textbox component as it's child", () => {
+      wrapper = renderPasswordInput({});
+      expect(wrapper.find(Textbox)).toHaveLength(1);
+    });
+  });
+
+  describe("required", () => {
+    beforeAll(() => {
+      wrapper = renderPasswordInput({ label: "required", required: true });
+    });
+
+    it("the required prop is passed to the input", () => {
+      input = wrapper.find("input").at(0);
+      expect(input.prop("required")).toBe(true);
+    });
+
+    it("the isRequired prop is passed to the label", () => {
+      const label = wrapper.find(Label).at(0);
+      expect(label.prop("isRequired")).toBe(true);
+    });
+  });
+
+  describe("password prop checks", () => {
+    it("default input type is `password`", () => {
+      expect(wrapper.find("input").at(0).prop("type")).toBe("password");
+    });
+
+    it("input type should change from `password` to `text` on click", () => {
+      act(() => {
+        wrapper.find(StyledButtonMinor).simulate("click", mockEvent);
+      });
+      wrapper.update();
+      expect(wrapper.find("input").at(0).prop("type")).toBe("text");
+    });
+
+    it("autoComplete attribute is `off`", () => {
+      expect(wrapper.find("input").at(0).prop("autoComplete")).toBe("off");
+    });
+
+    it.each([
+      ["foo", "foo"],
+      [mockedGuid, ""],
+    ])(
+      "Password input id should be custom value or guid depending on passed value",
+      (idType, id) => {
+        wrapper = renderPasswordInput({ id });
+        expect(wrapper.find("input").at(0).prop("id")).toBe(idType);
+      }
+    );
+  });
```

Comment:
Since we are testing the default behaviour and adding a custom id, I would perhaps split this test into two separate ones, just so if one test fails it has a clearer description of what it is trying to verify, which would help when debugging:

```tsx
it("input id should equal value of `id` prop when passed", () => {/**/});
it("input id should be a generated guid when `id` prop is undefined", () => {/**/});
```

Reply:
Yeah agreed